File: Projekt_Python/rosliny/Barszcz.py
import random
from Roslina import Roslina
from Zwierze import Zwierze
from zwierzeta.CyberOwca import CyberOwca


class Barszcz(Roslina):
    def __init__(self, x, y, dane):
        if dane == None:
            self._x = x
            self._y = y
            self._symbol = 'b'
            self._sila = 0
            self._wykonal_ruch = False
            self._zycie = 0
            self._inicjatywa = 0
            self._numer = 4
        else:
            self._x = dane[0]
            self._y = dane[1]
            self._inicjatywa = dane[2]
            self._symbol = 'b'
            self._sila = dane[3]
            self._wykonal_ruch = False
            self._zycie = dane[4]
            self._numer = dane[5]
        self._kolor = (0, 0, 0)

    def kolizja(self, organizm, organizmy, j, swiat):
        if isinstance(organizm, Zwierze):
            if organizm.getSymbol() == 'Q' and organizm.getX()==self._x and organizm.getY()== self._y:
                swiat.setRaport("b ginie od "+organizm.getSymbol())
                swiat.setPlanszaKolizja(organizm.getX(), organizm.getY(),organizm)
                organizmy.remove(self)
            elif organizm.getSymbol() != 'Q':
                swiat.setRaport(organizm.getSymbol() + " ginie od b ")
                swiat.setPlanszaNone(organizm.getX(), organizm.getY())
                organizmy.remove(organizm)
            j-=1

    def wykonajRuch(self, swiat, organizmy, j):
        kierunek = random.randint(0, 3)
        sum = 0
        if swiat.czyWPlanszy(self._x + 1, self._y):
            if swiat.getPlansza(self._x + 1, self._y) != None:
                self.kolizja(swiat.getPlansza(self._x + 1, self._y), organizmy, j, swiat)
                sum += 1
        else:
            sum += 1
        if swiat.czyWPlanszy(self._x - 1, self._y):
            if swiat.getPlansza(self._x - 1, self._y) != None:
                self.kolizja(swiat.getPlansza(self._x - 1, self._y), organizmy, j, swiat)
                sum += 1
        else:
            sum += 1
        if swiat.czyWPlanszy(self._x, self._y + 1):
            if swiat.getPlansza(self._x, self._y + 1) != None:
                self.kolizja(swiat.getPlansza(self._x, self._y + 1), organizmy, j, swiat)
                sum += 1
        else:
            sum += 1
        if swiat.czyWPlanszy(self._x, self._y - 1):
            if swiat.getPlansza(self._x, self._y - 1) != None:
                self.kolizja(swiat.getPlansza(self._x, self._y - 1), organizmy, j, swiat)
                sum += 1
        else:
            sum += 1
        if sum == 4: return

        if kierunek == 0:
            while True:
                xact = self._x
                yact = self._y
                kierunek = random.randint(0, 3)
                if kierunek == 0:  # //^
                    xact -= 1
                if kierunek == 1:  # //>
                    yact += 1
                if kierunek == 2:  # //v
                    xact += 1
                if kierunek == 3:  # //<
                    yact -= 1

                if swiat.czyWPlanszy(xact, yact) == False: continue
                org = swiat.getPlansza(xact, yact)
                if org == None:
                    if swiat.czyWPlanszy(self._x + 1, self._y):
                        if swiat.getPlansza(self._x + 1, self._y) != None:
                            self.kolizja(swiat.getPlansza(self._x + 1, self._y), organizmy, j, swiat)

                    if swiat.czyWPlanszy(self._x - 1, self._y):
                        if swiat.getPlansza(self._x - 1, self._y) != None:
                            self.kolizja(swiat.getPlansza(self._x - 1, self._y), organizmy, j, swiat)

                    if swiat.czyWPlanszy(self._x, self._y + 1):
                        if swiat.getPlansza(self._x, self._y + 1) != None:
                            self.kolizja(swiat.getPlansza(self._x, self._y + 1), organizmy, j, swiat)

                    if swiat.czyWPlanszy(self._x, self._y - 1):
                        if swiat.getPlansza(self._x, self._y - 1) != None:
                            self.kolizja(swiat.getPlansza(self._x, self._y - 1), organizmy, j, swiat)
                    break

            swiat.tworzRosline(self._numer, xact, yact)
            swiat.setRaport("Rozrasta " + self._symbol)

File: projekt_java/src/po/java/rosliny/Barszcz.java
package po.java.rosliny;

import po.java.Organizm;
import po.java.Roslina;
import po.java.Swiat;
import po.java.Zwierze;

import java.awt.*;
import java.util.ArrayList;
import java.util.Random;

public class Barszcz extends Roslina {
    public Barszcz(int x, int y) {
        this.x = x;
        this.y = y;
        this.symbol = 'b';
        this.sila = 0;
        this.wykonal_ruch = false;
        this.zycie = 0;
        this.numer=4;
       // this.kolor="black";
        this.kolor= Color.BLACK;
    }
    public Barszcz(int []dane)
    {
        this.x = dane[0];
        this.y = dane[1];
        this.inicjatywa=dane[2];
        this.symbol = 'b';
        this.sila = dane[3];
        this.wykonal_ruch = false;
        this.zycie = dane[4];
        this.numer=dane[5];
        this.kolor= Color.BLACK;
    }
    @Override
    public void kolizja(Organizm organizm, ArrayList organizmy, int j, Swiat swiat) {
        if (organizm instanceof Zwierze)
        {
            swiat.setRaport(organizm.getSymbol()+" ginie od b ");
            swiat.setPlansza(organizm.getX(), organizm.getY());
            organizmy.remove(organizm);
            j--;
        }
    }
    @Override
    public void wykonajRuch(Swiat swiat, ArrayList organizmy, int j) {
        Random liczba= new Random();
        int kierunek = liczba.nextInt(4);
        int xact = x, yact = y;
        int sum = 0;
        Organizm org;
        if (swiat.czyWPlanszy(swiat, x + 1, y)) {
            if (swiat.getPlansza(x + 1, y) != null)
            {
                kolizja(swiat.getPlansza(x+1,y ), organizmy,j,swiat);
                sum++;
            }
        }
        else sum++;
        if (swiat.czyWPlanszy(swiat, x - 1, y))
        {
            if (swiat.getPlansza(x - 1, y) != null) {
                kolizja(swiat.getPlansza(x-1,y ), organizmy,j,swiat);
                sum++;
            }
        }
        else sum++;
        if (swiat.czyWPlanszy(swiat, x, y+1)) {
            if (swiat.getPlansza(x, y+1) != null) {
                kolizja(swiat.getPlansza(x,y+1 ), organizmy,j,swiat);
                sum++;
            }
        }
        else sum++;
        if (swiat.czyWPlanszy(swiat, x, y-1)) {
            if (swiat.getPlansza(x, y-1) != null) {
                kolizja(swiat.getPlansza(x,y-1 ), organizmy,j,swiat);
                sum++;
            }
        }
        else sum++;
        if (sum==4) return;

        if (kierunek == 0)
        {
            do
            {
                xact = x;
                yact = y;
                kierunek = liczba.nextInt(4);
                if (kierunek == 0)//^
                {
                    xact--;
                }
                if (kierunek == 1)//>
                {
                    yact++;
                }
                if (kierunek == 2)//v
                {
                    xact++;
                }
                if (kierunek == 3)//<
                {
                    yact--;
                }
                if (!swiat.czyWPlanszy(swiat, xact, yact)) continue;
                org = swiat.getPlansza(xact, yact);
                if (org == null) {
                    if (swiat.czyWPlanszy(swiat,x+1,y))
                    {
                        if (swiat.getPlansza(x + 1,y) != null)
                        {
                            kolizja(swiat.getPlansza(x + 1,y), organizmy,j,swiat);
                        }
                    }
                    if (swiat.czyWPlanszy(swiat, x - 1, y))
                    {
                        if (swiat.getPlansza(x - 1,y) != null)
                        {
                           kolizja(swiat.getPlansza(x - 1,y), organizmy,j,swiat);
                        }
                    }
                    if (swiat.czyWPlanszy(swiat, x, y+1))
                    {
                        if (swiat.getPlansza(x,y + 1) != null)
                        {
                            kolizja(swiat.getPlansza(x,y + 1),organizmy,j,swiat);
                        }
                    }
                    if (swiat.czyWPlanszy(swiat, x, y-1))
                    {
                        if (swiat.getPlansza(x,y - 1) != null)
                        {
                            kolizja(swiat.getPlansza(x,y - 1), organizmy,j,swiat);
                        }
                    }
                    break;
                }
            } while (true);
            swiat.tworzRosline(numer,xact,yact);
            swiat.setRaport("Rozrasta "+symbol);
        }
    }
}
